ecosystem/supersim.md

```diff
@@ -0,0 +1,319 @@
+           D1[CLI Commands] -->|Command Line| D1a[start]
+           D1 -->|Command Line| D1b[stop]
+           D1 -->|Command Line| D1c[restart]
+          
+           D2[UI Buttons] -->|Graphical Interface| D2a[start]
+           D2 -->|Graphical Interface| D2b[stop]
+           D2 -->|Graphical Interface| D2c[restart]
+       end
+   end
+```
+
+```mermaid
+flowchart TD
+  subgraph Op-Simulator Interface
+      direction LR
+
+
+      C -->|any other RPC call| D2[call eth_sendRawTransaction \non Anvil]
+
+
+      C[ETH RPC API] -->|eth_sendRawTransaction| C1[Does TX have \n ExecutingMessage?]
+             C1 -->|no| D2
+
+
+      C1 -->|yes| D1
+
+
+    
+          D1[Check Interop Message Invariants] -->|If Pass|D2
+          D1 -->|If Fail| D3[Return TX failed]
+  
+    
+      D2 --> B3[Return result]
+  end
+```
+
+### Services
+
+Given the application flow, the following services will be part of supersim: 
+
+1. **Orchestrator**: This is the main focus and interface for developers using supersim. 
+    
+    * Runs all of the other services in supersim and manages their lifecycle
+    * Provides an interface to the user
+    
+        * Initially a static interface.
+        * Medium-term, offer an API interface that the orchestrator exposes to make it easier to build new offchain services that hook into the runtime.
+
+2. **Anvil Instances**: Anvil instances are run by the orchestrator after initial prep work to allow them to mirror the Superchain interop environment.
+
+    * Should be run in `--optimism` mode so the extra fields in transactions are there.
+    * Orchestrator should also run an L1 instance, though L1 withdrawals won’t initially be supported.
+    * Eventually supersim can be extended to be used with HardHat and other local simulation tools as well.
+
+3. **Genesis-applier** : The `genesis-applier` takes genesis files output by the monorepo and applies them to anvil instances.
+
+    * The genesis files needed will differ depending on whether the user is in vanilla or forked mode, and forked mode is unlikely to need anything other than the interop contracts initially. 
+    * Can be multi-step: multiple genesis files can be applied idempotently, particularly in forking mode (when you fork a network that has bedrock contracts but not interop contracts, for instance).
+    * Genesis-applier should work as a standalone tool: if a developer is using Anvil outside the context of the orchestrator, they should be able to use genesis-applier to apply genesis state to their instance. In the context of supersim, the `orchestrator` will use the `genesis-applier` to apply the needed genesis state to the anvil instances it is running. 
+
+4. **Chain-config-loader**: Downloads the latest chain config from the superchain registry to allow the orchestrator to apply needed configs to anvil instances. Should be a standalone tool that works with the orchestrator but could be used directly on local testing nodes. 
```

Comment:
If it's part of the other tooling, and written in Go or rust, then it should be relatively straight-forward to just import the superchain-registry as library. Like the op-node does.